### PR TITLE
feat(ELB): elb loadbalancer add params

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  elb loadbalancer add params bandwidth_id, backend_subnets, protection_status, protection_reason and add attribute ipv4_port_id
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  elb loadbalancer add params backend_subnets, protection_status, protection_reason and add attribute ipv4_port_id
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/elb/' TESTARGS='-run TestAccElbV3LoadBalancer_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/elb/ -v -run TestAccElbV3LoadBalancer_ -timeout 360m -parallel 4
=== RUN   TestAccElbV3LoadBalancer_basic
=== PAUSE TestAccElbV3LoadBalancer_basic
=== RUN   TestAccElbV3LoadBalancer_withEpsId
=== PAUSE TestAccElbV3LoadBalancer_withEpsId
=== RUN   TestAccElbV3LoadBalancer_withEIP
=== PAUSE TestAccElbV3LoadBalancer_withEIP
=== RUN   TestAccElbV3LoadBalancer_withEIP_Bandwidth_Id
=== PAUSE TestAccElbV3LoadBalancer_withEIP_Bandwidth_Id
=== RUN   TestAccElbV3LoadBalancer_prePaid
=== PAUSE TestAccElbV3LoadBalancer_prePaid
=== CONT  TestAccElbV3LoadBalancer_basic
=== CONT  TestAccElbV3LoadBalancer_prePaid
=== CONT  TestAccElbV3LoadBalancer_withEIP
=== CONT  TestAccElbV3LoadBalancer_withEpsId
=== CONT  TestAccElbV3LoadBalancer_prePaid
    acceptance.go:440: This environment does not support prepaid tests
--- SKIP: TestAccElbV3LoadBalancer_prePaid (0.00s)
=== CONT  TestAccElbV3LoadBalancer_withEIP_Bandwidth_Id
--- PASS: TestAccElbV3LoadBalancer_withEpsId (30.30s)
--- PASS: TestAccElbV3LoadBalancer_withEIP (34.35s)
--- PASS: TestAccElbV3LoadBalancer_withEIP_Bandwidth_Id (41.43s)
--- PASS: TestAccElbV3LoadBalancer_basic (110.57s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/elb       110.634s
```

### CODE REVIEW

File: docs/resources/elb_loadbalancer.md [hunk 1]
```diff
@@ -131,10 +131,31 @@ The following arguments are supported:
...
 * `l7_flavor_id` - (Optional, String) The L7 flavor id of the load balancer.
 
+* `backend_subnets` - (Optional, List) The IDs of subnets on the downstream plane.
+    If this parameter is not specified, select subnets as follows:
+      If IPv6 is enabled for a load balancer, the ID of subnet specified in `ipv6_network_id` will be used.
+      If IPv4 is enabled for a load balancer, the ID of subnet specified in `ipv4_subnet_id` will be used.
+      If only public network is available for a load balancer, the ID of any subnet in the VPC where the load balancer
+      resides will be used. Subnets with more IP addresses are preferred.
+      If there is more than one subnet, the first subnet in the list will be used, and the subnets must be in the VPC
+      where the load balancer resides.
```
Comment (deer-hang) [inline docs/resources/elb_loadbalancer.md]: Leave two spaces before the line

File: docs/resources/elb_loadbalancer.md [hunk 2]
```diff
@@ -166,6 +187,7 @@ The following arguments are supported:
 
 In addition to all arguments above, the following attributes are exported:
 
+* `ipv4_port_id` - The ID of the port bound to the private IPv4 address of the load balancer.
```
Comment (deer-hang) [inline docs/resources/elb_loadbalancer.md]:
```suggestion
* `ipv4_port_id` - The ID of the port bound to the private IPv4 address of the Load Balancer.
```

File: huaweicloud/services/elb/resource_huaweicloud_elb_loadbalancer.go [hunk 3]
```diff
@@ -98,18 +98,15 @@ func ResourceLoadBalancerV3() *schema.Resource {
 				Computed: true,
 				ForceNew: true,
 				ConflictsWith: []string{
-					"iptype", "bandwidth_charge_mode", "bandwidth_size", "sharetype",
+					"iptype", "bandwidth_charge_mode", "bandwidth_size", "sharetype", "bandwidth_id",
```
Comment (deer-hang) [inline huaweicloud/services/elb/resource_huaweicloud_elb_loadbalancer.go:101]: Please change the document together(document line 123).
And check the others.

File: huaweicloud/services/elb/resource_huaweicloud_elb_loadbalancer.go [hunk 4]
```diff
@@ -434,6 +478,31 @@ func resourceLoadBalancerV3Update(ctx context.Context, d *schema.ResourceData, m
...
 
+	// backend_subnets and cross_vpc_backend can not be updated at the same time, otherwise, an error will occur
+	if d.HasChange("backend_subnets") {
+		updateOpts := loadbalancers.UpdateOpts{
+			ElbSubnetIds: utils.ExpandToStringList(d.Get("backend_subnets").(*schema.Set).List()),
+		}
+		// if the value of vip_subnet_cidr_id and ipv6_vip_virsubnet_id are null, then they will be unbound
+		if v, ok := d.GetOk("ipv4_subnet_id"); ok {
+			vipSubnetID := v.(string)
+			updateOpts.VipSubnetID = &vipSubnetID
```
Comment (deer-hang) [inline huaweicloud/services/elb/resource_huaweicloud_elb_loadbalancer.go]:
```suggestion
			updateOpts.VipSubnetID = utils.String(v.(string))
```
Please check the others.

File: huaweicloud/services/elb/resource_huaweicloud_elb_loadbalancer.go [hunk 5]
```diff
@@ -143,7 +140,18 @@ func ResourceLoadBalancerV3() *schema.Resource {
...
+
+			"bandwidth_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				RequiredWith: []string{
+					"iptype",
+				},
+				ConflictsWith: []string{"ipv4_eip_id", "bandwidth_size", "bandwidth_charge_mode", "sharetype"},
```
Comment (deer-hang) [inline huaweicloud/services/elb/resource_huaweicloud_elb_loadbalancer.go]: If the sharetype is set to WHOLE,  the bandwidth_id is required.

File: docs/resources/elb_loadbalancer.md (hunk 1 above)
Comment (houpeng80) [reply docs/resources/elb_loadbalancer.md]: done

File: docs/resources/elb_loadbalancer.md (hunk 2 above)
Comment (houpeng80) [reply docs/resources/elb_loadbalancer.md]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_loadbalancer.go (hunk 3 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_loadbalancer.go:101]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_loadbalancer.go (hunk 4 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_loadbalancer.go]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_loadbalancer.go (hunk 5 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_loadbalancer.go]: done

File: huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_loadbalancer_test.go [hunk 6]
```diff
@@ -288,6 +349,33 @@ resource "huaweicloud_elb_loadbalancer" "test" {
...
+resource "huaweicloud_elb_loadbalancer" "test" {
+  name           = "%[1]s"
+  ipv4_subnet_id = data.huaweicloud_vpc_subnet.test.ipv4_subnet_id
+
+  availability_zone = [
+    data.huaweicloud_availability_zones.test.names[0]
+  ]
+
+  iptype       = "5_bgp"
+  bandwidth_id = huaweicloud_vpc_bandwidth.test.id
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/elb/resource_huaweicloud_elb_loadbalancer_test.go:375]: Please add a test with share_type="WHOLE" with bandwidth_id is not empty

**Comment by houpeng80** [comment]
make testacc TEST='./huaweicloud/services/acceptance/elb/' TESTARGS='-run TestAccElbV3LoadBalancer_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/elb/ -v -run TestAccElbV3LoadBalancer_ -timeout 360m -parallel 4
=== RUN   TestAccElbV3LoadBalancer_basic
=== PAUSE TestAccElbV3LoadBalancer_basic
=== RUN   TestAccElbV3LoadBalancer_withEpsId
=== PAUSE TestAccElbV3LoadBalancer_withEpsId
=== RUN   TestAccElbV3LoadBalancer_withEIP
=== PAUSE TestAccElbV3LoadBalancer_withEIP
=== RUN   TestAccElbV3LoadBalancer_withEIP_Bandwidth_Id
=== PAUSE TestAccElbV3LoadBalancer_withEIP_Bandwidth_Id
=== RUN   TestAccElbV3LoadBalancer_prePaid
=== PAUSE TestAccElbV3LoadBalancer_prePaid
=== CONT  TestAccElbV3LoadBalancer_basic
=== CONT  TestAccElbV3LoadBalancer_withEIP_Bandwidth_Id
=== CONT  TestAccElbV3LoadBalancer_withEIP
=== CONT  TestAccElbV3LoadBalancer_withEpsId
--- PASS: TestAccElbV3LoadBalancer_withEpsId (29.48s)
=== CONT  TestAccElbV3LoadBalancer_prePaid
    acceptance.go:440: This environment does not support prepaid tests
--- SKIP: TestAccElbV3LoadBalancer_prePaid (0.01s)
--- PASS: TestAccElbV3LoadBalancer_withEIP (31.78s)
--- PASS: TestAccElbV3LoadBalancer_withEIP_Bandwidth_Id (38.65s)
--- PASS: TestAccElbV3LoadBalancer_basic (105.85s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/elb       158.693s

File: docs/resources/elb_loadbalancer.md [hunk 7]
```diff
@@ -120,21 +120,55 @@ The following arguments are supported:
...
+  -> **NOTE:** If the `bandwidth_id` parameter is configured, you can not configure the parameters:
+  `bandwidth_charge_mode`, `bandwidth_size`.
 
 * `l4_flavor_id` - (Optional, String) The L4 flavor id of the load balancer.
 
 * `l7_flavor_id` - (Optional, String) The L7 flavor id of the load balancer.
 
+* `backend_subnets` - (Optional, List) The IDs of subnets on the downstream plane.
+  If this parameter is not specified, select subnets as follows:
+  If IPv6 is enabled for a load balancer, the ID of subnet specified in `ipv6_network_id` will be used.
```
Comment (ShiChangkuo) [inline docs/resources/elb_loadbalancer.md]: it's better to use list item
```suggestion
  + If IPv6 is enabled for a load balancer, the ID of subnet specified in `ipv6_network_id` will be used.
```
